File: epi_judge_python/is_number_palindromic.py
from test_framework import generic_test
import math


def is_palindrome_number(x: int) -> bool:
    if x < 0:
        return False
    if x == 0:
        return True
    dlen = math.floor(math.log10(x))
    msd = dlen
    while x > 0:
        md = x // (10 ** msd)
        if md != x % 10:
            return False
        x -= md * (10 ** msd)
        msd -= 2
        x //= 10
        
    return True


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('is_number_palindromic.py',
                                       'is_number_palindromic.tsv',
                                       is_palindrome_number))

File: epi_judge_cpp/is_number_palindromic.cc
#include "test_framework/generic_test.h"
bool IsPalindromeNumber(int x) {
  // TODO - you fill in here.
  // return true;

  if (x < 0) {
    return false;
  }

  long long reversed{0};
  int temp{x};
  while (x) {
    reversed *= 10;
    reversed += (x % 10);
    x /= 10;
  }
  return (reversed == temp) ? true : false;
}

int main(int argc, char *argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"x"};
  return GenericTestMain(args, "is_number_palindromic.cc",
                         "is_number_palindromic.tsv", &IsPalindromeNumber,
                         DefaultComparator{}, param_names);
}
